### PR TITLE
Open at login setting

### PR DESCRIPTION
**Pull request by gabriel**
Electron support for login items has improved a ton!
https://github.com/electron/electron/blob/master/docs/api/app.md#appsetloginitemsettingssettings-macos-windows

- Sets system preferences login item on first run (if it has never been set and is installed in /Applications)
- Uses system preferences login item setting whether to start on boot instead of "app start mode" from service
- Restores UI if opened from login (boot)
- Hides window & dock if login item is set to open as hidden
- Removes app_start_mode setting and all code associated with it

A lot of manual testing.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

**Comment by gabriel** [comment]
2016-08-16 at 5 51 pm

**Comment by gabriel** [comment]
And app start mode was really annoying so it's good that's gone (cc @maxtaco)

File: desktop/app/app-state.js [hunk 1]
```diff
@@ -60,6 +62,18 @@ export default class AppState {
     })
   }
 
+  checkOpenAtLogin () {
+    console.log('Checking open at login')
```
Comment (MarcoPolo) [inline desktop/app/app-state.js]: ✂️

Comment (MarcoPolo) [inline desktop/app/app-state.js]: (same with other console.logs, unless you really want them)

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**